### ISSUE TITLE
[1.7.7.5 and 1.7.6.1] [Multistore] Product is removed from search index when saved after edit

### ISSUE THREAD

**Issue by davidcylke** [PrestaShop/PrestaShop#25512]
When we edit the product in "All stores" (for example add some features) and save the product its removed from search index on one of the versions of our multistore. We had same issue in 1.7.6.1 and now in 1.7.7.5. Its very annoying, because preferences - search says all products are indexed. Adding only missing ones button doesn't do anything and only rebuilding whole index fixes it, while we have 12k products, so its time consuming.

#### Expected behavior

Product shouldn't be removed from search index when saved.

#### Steps to Reproduce

1. Go to product edit when in "All stores" multistore option
2. Click on add feature, add some feature and Save product.
3. Go to front office on both, b2b and b2c versions - its not findable on one of the sites.
4. Get frustrated.

Screen recording shows how it works.

https://www.icloud.com/iclouddrive/0Nrm0y0BRK0I3efMfCWcJK51Q#Nagranie_z_ekranu_2021-08-4_o_21.31
#### Additional information

* PrestaShop version: 1.7.7.5
* PHP version: 7.2

**Comment by hibatallahAouadni** [comment]
Hello @davidcylke 

I think you have the same issue as this one #17778
I didn't manage to reprodcue the issue with PS1775 and PS1776, see the attached screen record below:

https://drive.google.com/file/d/1xV6FakJBISj5G0VAx4aCfuNVpbrxj2FK/view

And could you please check this [comment](https://github.com/PrestaShop/PrestaShop/issues/17778#issuecomment-589726852) :pray: 
Please check and feedback.

Thanks!

**Comment by davidcylke** [comment]
No, its not the same issue. Adding missing products to the index is not changing anything.

Also on you video you didn't work on "All shops".

**Comment by davidcylke** [comment]
Also, on the video you didn't use search on the front office. In my case search index says there are all products indexes but the edited products is not searchable

**Comment by khouloudbelguith** [comment]
Hi @davidcylke,

I did not manage to reproduce the issue with PS1.7.7.5.
I attached a screen record
https://drive.google.com/file/d/1a7ZvPXPHs3KrTZTd4XYH8YdZyKboYvdk/view?usp=sharing

Also, could you please heck in the BO > Product details > Options Tab > Visibility = Everywhere.

Thanks!

**Comment by davidcylke** [comment]
Yeah, I have Everywhere set.

**Comment by khouloudbelguith** [comment]
@davidcylke, have you any custom modules related to the search? or it could be to your custom theme?
If no, what is the exact configuration of the Shop Group?

Thanks!
